src/v/cluster/topics_frontend.cc

```diff
@@ -1284,4 +1284,41 @@ ss::future<std::error_code> topics_frontend::decrease_replication_factor(
       _stm, _features, _as, std::move(cmd), timeout);
 }
 
+ss::future<bool> topics_frontend::replica_set_exists(
+  model::ntp ntp, std::vector<model::node_id> replicas) {
+    auto current_assignment = _topics.local().get_partition_assignment(ntp);
+    if (!current_assignment.has_value()) {
+        // No current assignment, so the replica set does not exist
+        co_return false;
+    }
+
+    if (current_assignment->replicas.size() != replicas.size()) {
+        // Unequal sizes, so the replica set does not exist
+        co_return false;
+    }
+
+    for (const model::broker_shard& bshard : current_assignment->replicas) {
+        auto broker_in_replica_set_it = std::find(
+          replicas.begin(), replicas.end(), bshard.node_id);
+        if (broker_in_replica_set_it == replicas.end()) {
+            co_return false;
+        }
+    }
+
+    for (const model::node_id& node_id : replicas) {
```

Comment:
if we assume theres no duplicates in these vectors, knowing 1. that they are equal in their sizes and 2. all node_ids from collection A are in collection B, then there's no need to perform this additional loop checking that node_ids from collection B are in collection A.

Reply:
I think a more succinct solution would be to sort the arrays or put the node_ids into two respective sets and `return a == b`. Advantages are, less code, and being more obvious to reader what is going on.